pkg/controllers/customeradmin/group_mapping.go

```diff
@@ -33,7 +35,7 @@ func fromMSGraphGroup(log *logrus.Entry, kubeGroup *v1.Group, kubeGroupName stri
 	g.Users = []string{}
 	for _, user := range msGroupMembers {
 		if user.UserType == graphrbac.Guest && user.Mail != nil {
-			g.Users = append(g.Users, *user.Mail) // doesn't include #EXT#
```

Comment:
foo := reconcileGuestUsers(log, userV1, user)
if foo != "" {
g.Users = append(g.Users, foo)
}
This will prevent useless entries in the group

Reply:
I think the current logic is fine. we still want the record to be added if we reached that point of code. Its just question of "if it was a prefix or no". Or I am missing something?

Reply:
We don't validate within `reconcileGuestUsers` that `*AADUser.Mail` is non-empty so I think Angus has a decent point here. If you want to golf it even more I think you can write:

```go
if reconciled := reconcileGuestUsers(log, userV1, user); reconciled != "" {
    g.Users = append(g.Users, reconciled)
}
```

Reply:
> I think the current logic is fine. we still want the record to be added if we reached that point of code. Its just question of "if it was a prefix or no". Or I am missing something?

user.Mail as a member of the group does nothing, correct? That's why we are fixing this bug...
I am just suggesting not to add it as it just clutters up the group list and will be confusing.
So if we have a group with lots of guests, but they haven't logged in we will have lots of these non-functional members (and they don't get cleaned up when the user does login).
Then only useful entry is the new one that you are adding "usr.Name" (which includes the foo.com#). Anyways, that is my understanding..

Reply:
We have a different type of guest users. in example, `@microsfot.com` email does not have any prefix. I assume MSFT trust their own accounts. I suspect similar behaviour will be with live.com and outlook.com accounts. 

And I would like to avoid this "I signed in but still can't see projects, and they appeared after 1 minute". So if user configured 10 users in Group we move all 10 into the group. And just modify the group if some of those users appear to be different (with prefix) post login.

Reply:
Otherwise all this delays user sync. And not sure if this is what we would like to do?

Reply:
ok understood @mjudeikis I thought all had a prefix. lgtm..

Reply:
not all. And even more fun. Guest accounts from @microsoft.com or @live.com in your domain tree does not have this..